### ISSUE TITLE
error that 7.0 .NET SDK is being used, not 8.0

### ISSUE THREAD

**Issue by danmoseley** [dependabot/dependabot-core#8530]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

nuget

### Package manager version

whatever is in the 8.0 SDK

### Manifest location and content before the Dependabot update

various in https://github.com/danmoseley/eShop/tree/dependabot

### dependabot.yml content

https://github.com/danmoseley/eShop/blob/dependabot/.github/dependabot.yml

### What you expected to see, versus what you actually saw

log: https://github.com/danmoseley/eShop/network/updates/758220044 

errors appear to start ere

```
updater |   Updating global.json files.
updater |     Dependency [Duende.IdentityServer.AspNetIdentity] not found in any global.json files.
updater |   No dotnet-tools.json files found.
updater | Running for project [/home/dependabot/dependabot-updater/repo/src/eShop.AppHost/eShop.AppHost.csproj]
updater |   Running for SDK-style project
updater | dotnet build in GetAllPackageDependenciesAsync failed. STDOUT: MSBuild version 17.6.3+07e294721 for .NET
updater |   Determining projects to restore...
updater | /usr/local/dotnet/current/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(151,5): error NETSDK1045: The current .NET SDK does not support targeting .NET 8.0.  Either target .NET 7.0 or lower, or use a version of the .NET SDK that supports .NET 8.0. [/tmp/package-dependency-resolution_66Ggzt/Project.csproj]
```

The .NET 8.0.100 SDK is shipped 3 weeks ago. As far as I know, my repo is calling for it -- the global.json has
```json
{
  "sdk": {
    "version": "8.0.100",
    "rollForward": "latestPatch",
    "allowPrerelease": true
  }
}
```

Is Dependabot using the 7.0 SDK still? That would cause it to fail when examining projects that target 8.0.

**Comment by danmoseley** [comment]
BTW I see this mentioned in https://github.com/dependabot/dependabot-core/issues/8483 by @martincostello but it's not clear to me that issue covers it.

**Comment by martincostello** [comment]
Yeah, it's a weird one. Every single repo I have where I use Dependabot and .NET has been upgraded to .NET 8, but only two of them are hitting this specific error.

**Comment by danmoseley** [comment]
> only two of them are hitting this specific error.

Could it be that dependabot is running on a pool of machines only some of which have it installed? One would have thought it would be in a consistent container though.

**Comment by Smalls1652** [comment]
Might be helpful to have the `dotnet-install` script install the latest version of each "in support" and preview release of the .NET SDK during the container image build. Side-by-side installs of .NET SDKs is possible.

I can see a potential problem with that though. IIRC the `dotnet` CLI will default to the highest version available, so if an in-development/preview release of the .NET 9 SDK is installed it will use that over the other versions. The only way to force it to use .NET 6/7/8 would be to have a `global.json` in the repo, but not everyone uses that and I can see that being a problem.

**Comment by martincostello** [comment]
FWIW, all my repos have a `global.json` for reproducibility.

**Comment by danmoseley** [comment]
I'll actually need .NET 9 support almost immediately, eg., dotnet/eshop (which is owned by the .NET team) will take a dependency on it right away. Clearly, we're outliers, but presumably if the problem is solved for any previews, it will work for us.

**Comment by danmoseley** [comment]
I think the ideal would be to allow a flag in the yml specifying that dependabot should use the latest preview .NET rather than latest stable. It's just a flag to the install script.

My guess is that it would be unlikely for dependabot to hit a bug introduced in a preview
as the dependabot use of .NET is not particularly extensive or stressful (?) but this would eliminate any risk.

Also as a .NET committer I would certainly champion fixes to any problem it may encounter (and @martincostello is a regular contributor there too.)
